### ISSUE TITLE
TF module doesn't work after running hardening level 1

### ISSUE THREAD

**Issue by tibers** [rancher/terraform-provider-rancher2#110]
Hi folks. I have a cluster I provisioned with the hardening guide, and I have a cluster I provisioned without the hardening guide. My terraform code works against the cluster provisioned **without** the hardening guide, but it does not work against the cluster provisioned **with** the hardening guide.

[Hardening Guide for Reference](https://rancher.com/docs/rancher/v2.x/en/security/hardening-2.1/)

My WAG is that something in the hardening guide (perhaps [anonymous auth](https://rancher.com/docs/rancher/v2.x/en/security/hardening-2.1/#2-1-2-configure-kube-api-options)) breaks something in rancher because perhaps the terraform module attempts to plan without an auth token, which is where my stuff dies very early on in the process.

The message back is always `Error: Bad response statusCode [401]. Status [401 Unauthorized]. Body: [message=must authenticate] from [https://rancher.mydomain.com/v3]`

When I test this with `docker run --net=host superseb/rancher2ws:latest rancher.us-east-1.aws.mydomain.com token-....` the token works so it's definitely something funky terraform is doing.

**Comment by rawmind0** [comment]
Hi @tibers,

Terraform provider should not plan without an auth token. Could you please provide you tf file?? Have you checked that api token provided on tf has permission granted to exec required actions on rancher??

**Comment by tibers** [comment]
The auth token is provisioned from a service account with admin privs in Azure AD. I have also tried logging in with the local admin account and provisioning it that way. Both result in 401s. I also set "Scope" to unset (no scope). 

Is there a better way to provision a token than through the UI?

**Comment by rawmind0** [comment]
Providing a local admin "no scope" token, are you also getting `401`?? Are you able to connect to the API with same token?? What are you trying to create with tf??

Beside UI, you can create tokens from API, https://<RANCHER_URL>/v3/tokens/

**Comment by tibers** [comment]
OK. This was awful. There's nothing wrong with the terraform module. The problem is that as I'm working on the repo and changing the rancher endpoint I am using, I forgot to clean up the s3 bucket hosting the state file. The result is that all the stuff which ends up in the TF state was now out of sync with my new cluster endpoint I was attempting to use. Upon careful inspection of the output and the configs, the 401s were generated by clusters which the TF module thought it was supposed to be managing - but already existed with different credentials from the original cluster. 

So tl;dr, I made a huge mess for myself by not realizing that I should be wiping my S3 buckets.
